crates/agglayer-clock/src/block.rs

```diff
@@ -0,0 +1,363 @@
+use std::{
+    num::NonZeroU64,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+};
+
+use ethers::providers::{Middleware, PubsubClient};
+use futures::StreamExt as _;
+use tokio::sync::broadcast;
+use tokio_util::sync::CancellationToken;
+use tracing::debug;
+#[cfg(not(test))]
+use tracing::error;
+
+use crate::{Clock, ClockRef, Error, Event, BROADCAST_CHANNEL_SIZE};
+
+/// Block based [`Clock`] implementation.
+pub struct BlockClock<P> {
+    provider: Arc<P>,
+    genesis_block: u64,
+    current_block: Arc<AtomicU64>,
```

Comment:
I found it not so straightforward to follow the occurrences to `block` that currently refer to two different values:

1. L1 block height
2. L1 block height minus genesis block

For example:
- `current_block_height` in `ClockRef` refers to 1
- `current_block` in `BlockClock` refers to 2
- `from_block` in `calculate_epoch_number(&self, from_block: u64)` refers to 2
- `from_block` in `calculate_block_number(&self, from_block: u64)` refers to 1
- `block_number` in `calculate_block_number(&self, from_block: u64)` refers to 2

The easiest might be to keep 2. as is but to consistently prefix the references to 1. like `l1_block`? Alternatively using a different time unit for 2. like tick, slot, beat etc?

Reply:
`current_block` and `current_block_height` are the same, and they both refer to `2`.